build_system/src/test.rs

```diff
@@ -856,134 +856,135 @@ fn should_remove_test(file_path: &Path) -> Result<bool, String> {
+        fn dir_handling(dir: &Path) -> Result<(), String> {
+            if dir.file_name().map(|name| name == "auxiliary").unwrap_or(true) {
+                return Ok(());
+            }
+            walk_dir(dir, dir_handling, file_handling)
+        }
+        fn file_handling(file_path: &Path) -> Result<(), String> {
+            if !file_path.extension().map(|extension| extension == "rs").unwrap_or(false) {
+                return Ok(());
+            }
+            let path_str = file_path.display().to_string().replace("\\", "/");
+            if should_not_remove_test(&path_str) {
+                return Ok(());
+            } else if should_remove_test(file_path)? {
+                return remove_file(&file_path);
+            }
+            Ok(())
+        }
 
-    let nb_parts = args.nb_parts.unwrap_or(0);
-    if nb_parts > 0 {
-        let current_part = args.current_part.unwrap();
-        // FIXME: create a function "display_if_not_quiet" or something along the line.
-        println!("Splitting ui_test into {} parts (and running part {})", nb_parts, current_part);
-        let out = String::from_utf8(
-            run_command(
-                &[
-                    &"find",
-                    &"tests/ui",
-                    &"-type",
-                    &"f",
-                    &"-name",
-                    &"*.rs",
-                    &"-not",
-                    &"-path",
-                    &"*/auxiliary/*",
-                ],
-                Some(&rust_path),
-            )?
-            .stdout,
-        )
-        .map_err(|error| format!("Failed to retrieve output of find command: {:?}", error))?;
-        let mut files = out
-            .split('\n')
-            .map(|line| line.trim())
-            .filter(|line| !line.is_empty())
-            .collect::<Vec<_>>();
-        // To ensure it'll be always the same sub files, we sort the content.
-        files.sort();
-        // We increment the number of tests by one because if this is an odd number, we would skip
-        // one test.
-        let count = files.len() / nb_parts + 1;
-        let start = current_part * count;
-        // We remove the files we don't want to test.
-        for path in files.iter().skip(start).take(count) {
-            remove_file(&rust_path.join(path))?;
+        remove_file(&rust_path.join("tests/ui/consts/const_cmp_type_id.rs"))?;
+        remove_file(&rust_path.join("tests/ui/consts/issue-73976-monomorphic.rs"))?;
+        // this test is oom-killed in the CI.
+        remove_file(&rust_path.join("tests/ui/consts/issue-miri-1910.rs"))?;
+        // Tests generating errors.
+        remove_file(&rust_path.join("tests/ui/consts/issue-94675.rs"))?;
+        remove_file(&rust_path.join("tests/ui/mir/mir_heavy_promoted.rs"))?;
+        remove_file(&rust_path.join("tests/ui/rfcs/rfc-2632-const-trait-impl/const-drop-fail.rs"))?;
+        remove_file(&rust_path.join("tests/ui/rfcs/rfc-2632-const-trait-impl/const-drop.rs"))?;
```

Comment:
Can these be added to failing-ui-tests file. It would be better, not sure why they are here.

Reply:
Yes, I think we can move them. Or even better, it seems some of these tests have the annotation `// @known-bug` in them, so if this annotation is indeed indicating a known failure, we can add it [here](https://github.com/rust-lang/rustc_codegen_gcc/blob/master/build_system/src/test.rs#L914).